src/docs/backend/control-silo.mdx

```diff
@@ -0,0 +1,178 @@
+
+### Slug, id, and UUID based routing
+
+For requests that use `organization_slug`, `organization_id_or_slug`, `sentry_app_slug` or `uuid` parameters the API Gateway can resolve the organization’s region using data present in control silo. Once the region is resolved requests are synchronously proxied to the appropriate region.
+
+### DSN host
+
+The legacy error-embed page requires specialized handling, and we extract the
+destination from the `dsn` query string parameter.
+
+### Region Pin List
+
+We have a handful of API endpoints that lack useful hints to route requests between regions. When a request is made to one of these paths we assume the intent is to send a request to the `us` region.
+
+### Direct Location Header
+
+When a request is handled by the API Gateway, the response delivered to the client will contain `X-Sentry-Proxy-Url` header which contains the URL that clients should use to avoid the additional latency incurred by the API Gateway.
+
+# Integration Credential Proxy
+
+Integrations can be shared by multiple organizations. For example, a single MS teams workspace can be connected to multiple organizations, and those organizations can live in multiple regions. A few of our integrations leverage refresh tokens and use short lived access tokens. To avoid racy updates when tokens become invalidated and need to be refreshed, we send outbound requests to control silo, which acts as a proxy for the integration service. By funnelling all outbound traffic for integrations that require refresh tokens through control silo we can more easily co-ordinate token replacements.
+
+```mermaid
+sequenceDiagram
+
+actor RegionSilo
+actor ControlSilo
+actor MsTeams
+
+RegionSilo ->> RegionSilo: send issue alert
+RegionSilo ->> ControlSilo : send issue alert to msteams
+ControlSilo ->> ControlSilo : load integration metadata
+ControlSilo ->> ControlSilo : add integration credentials to request
+ControlSilo ->> MsTeams : send notification request with credentials
+MsTeams -->> ControlSilo : token expired!
+ControlSilo ->> MsTeams : refresh token
+MsTeams -->> ControlSilo : response
+ControlSilo ->> ControlSilo : update stored token
+ControlSilo ->> MsTeams : send notification with new token
+MsTeams -->> ControlSilo : response
+ControlSilo -->> RegionSilo : response
+```
+
+The integration proxy is implemented as a class that integrations requiring refresh tokens can sub-class. Currently the following integrations use the integration credential proxy:
+
+- GitLab
+- MSTeams
+- GitHub
+- Vsts (Azure Devops)
+
+### Integration Credential Proxy Headers
+
+The integration credential proxy leverages several specialized headers to perform its role:
+
+- `X-Sentry-Subnet-Organization-Integration` The id of the `OrganizationIntegration` that credentials should be added from.
+- `X-Sentry-Subnet-Base-URL` defines the integration host and base URL that the request should be forwarded to.
```

Comment:
```suggestion
- `X-Sentry-Subnet-Base-Url` defines the integration host and base URL that the request should be forwarded to.
```
Technically shouldn't matter but matches the general canonical naming for headers